### PR TITLE
[Merged by Bors] - chore: Golf `Set.pair_eq_pair_iff`

### PR DESCRIPTION
**Pull request by YaelDillies**
Resolve a porting note

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Basic.lean [hunk 1]
```diff
@@ -1373,14 +1373,9 @@ theorem pair_comm (a b : α) : ({a, b} : Set α) = {b, a} :=
...
 
--- Porting note: first branch after `constructor` used to be by `tauto!`.
 theorem pair_eq_pair_iff {x y z w : α} :
     ({x, y} : Set α) = {z, w} ↔ x = z ∧ y = w ∨ x = w ∧ y = z := by
-  simp only [Set.Subset.antisymm_iff, Set.insert_subset_iff, Set.mem_insert_iff,
-    Set.mem_singleton_iff, Set.singleton_subset_iff]
-  constructor
-  · rintro ⟨⟨rfl | rfl, rfl | rfl⟩, ⟨h₁, h₂⟩⟩ <;> simp [h₁, h₂] at * <;> simp [h₁, h₂]
-  · rintro (⟨rfl, rfl⟩ | ⟨rfl, rfl⟩) <;> simp
+  simp [subset_antisymm_iff, insert_subset_iff]; aesop
```
Comment (YaelDillies) [inline Mathlib/Data/Set/Basic.lean:1378]:
```suggestion
  aesop (add simp [subset_antisymm_iff, insert_subset_iff])
```
is twice as slow, cc @JLimperg

**Review by grunweg — APPROVED** [review]

**Comment by ADedecker** [comment]
Thanks! Delegating just in case you want to discuss the aesop issue before merging.

bors d+

**Comment by YaelDillies** [comment]
The aesop issue is non-blocking. Just wanted Jannis to have a chance to look!

maintainer merge

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**